### PR TITLE
Create proxy.json

### PR DESCRIPTION
**Pull request by hrasheed-msft**
Swagger specification for HDInsight Kafka REST proxy

### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/hdinsight/hdinsight-kafka-rest-proxy/proxy.json [hunk 1]
```diff
@@ -0,0 +1,496 @@
...
+        "consumes" : [ "text/plain" ],
+        "produces" : [ "application/json" ],
+        "parameters" : [ {
+          "name" : "topic",
+          "in" : "path",
+          "required" : true,
+          "type" : "string"
+        }, {
+          "in" : "body",
+          "name" : "body",
```
Comment (MyronFanQiu) [inline specification/hdinsight/hdinsight-kafka-rest-proxy/proxy.json]: Just want to confirm, is this name by design?

File: specification/hdinsight/hdinsight-kafka-rest-proxy/proxy.json [hunk 2]
```diff
@@ -0,0 +1,496 @@
...
+          }
+        }
+      }
+    }
+  },
+  "definitions" : {
+    "ConsumeRecord" : {
+      "type" : "object",
+      "properties" : {
+        "value" : {
```
Comment (MyronFanQiu) [inline specification/hdinsight/hdinsight-kafka-rest-proxy/proxy.json]: Could we add description?

**Comment by MyronFanQiu** [comment]
@hrasheed-msft Hello. Could you solve the CI error and address the comments?

**Comment by hrasheed-msft** [comment]
@MyronFanQiu I have reached out to the dev team that wrote the spec to resolve the issues. I'll update the PR shortly.

**Comment by hrasheed-msft** [comment]
@MyronFanQiu The team has updated the spec. Is there a way to exempt the terms causing the first spellcheck test to fail?

**Comment by MyronFanQiu** [comment]
@hrasheed-msft you can add your words into https://github.com/Azure/azure-rest-api-specs/blob/master/custom-words.txt

**Comment by hrasheed-msft** [comment]
> 
> 
> @hrasheed-msft you can add your words into https://github.com/Azure/azure-rest-api-specs/blob/master/custom-words.txt

I have added them in this PR: https://github.com/Azure/azure-rest-api-specs/pull/7855

File: specification/hdinsight/hdinsight-kafka-rest-proxy/proxy.json [hunk 3]
```diff
@@ -0,0 +1,599 @@
...
+          "type": "array",
+          "description": "List of partition metadata",
+          "items": {
+            "$ref": "#/definitions/RESTPartitionMetadataResponse"
+          }
+        }
+      },
+      "description": "Metadata about a specific topic"
+    },
+    "RestProxyBrokerEndpoint": {
```
Comment (devigned) [inline specification/hdinsight/hdinsight-kafka-rest-proxy/proxy.json]: `RestProxyBrokerEndpoint` is named differently than all the rest. For example:  `RESTTopicMetadaResponse`. Note the caps for REST.

File: specification/hdinsight/hdinsight-kafka-rest-proxy/proxy.json [hunk 4]
```diff
@@ -0,0 +1,599 @@
...
+          "description": "List of topic names",
+          "uniqueItems": true,
+          "items": {
+            "type": "string"
+          }
+        }
+      },
+      "description": "List of topics"
+    },
+    "RESTTopicMetadaResponse": {
```
Comment (devigned) [inline specification/hdinsight/hdinsight-kafka-rest-proxy/proxy.json]: Why prefix everything with REST? Seems like these names would not overlap with any other models in this namespace.

File: specification/hdinsight/hdinsight-kafka-rest-proxy/proxy.json [hunk 5]
```diff
@@ -0,0 +1,599 @@
...
+        "replicas": {
+          "type": "array",
+          "description": "List of replica broker endpoints for the topic-partition",
+          "items": {
+            "$ref": "#/definitions/RestProxyBrokerEndpoint"
+          }
+        },
+        "isr": {
+          "type": "array",
+          "description": "List of broker endpoints in ISR for the topic-partition",
```
Comment (devigned) [inline specification/hdinsight/hdinsight-kafka-rest-proxy/proxy.json]: Please expand the acronym `ISR`.

Comment (hrasheed-msft) [reply specification/hdinsight/hdinsight-kafka-rest-proxy/proxy.json]: fixed

File: specification/hdinsight/hdinsight-kafka-rest-proxy/proxy.json (hunk 4 above)
Comment (hrasheed-msft) [reply specification/hdinsight/hdinsight-kafka-rest-proxy/proxy.json]: fixed

File: specification/hdinsight/hdinsight-kafka-rest-proxy/proxy.json (hunk 3 above)
Comment (hrasheed-msft) [reply specification/hdinsight/hdinsight-kafka-rest-proxy/proxy.json]: fixed

**Comment by hrasheed-msft** [comment]
@MyronFanQiu @devigned The json file has been updated based on reviewer comments.

**Review by mmyyrroonn — APPROVED** [review]
LGTM @devigned Could you confirm whether you have more concerns about this PR or not?

**Comment by devigned** [comment]
👍

**Pull request merged**